src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -209,8 +275,8 @@ private void check() {
                         LOGGER.log(Level.FINE, "last-minute output in {0} on {1}", new Object[] {remote, node});
                     }
                     t.set(null); // do not interrupt cleanup
-                    if (exitCode == 0) {
-                        getContext().onSuccess(exitCode);
+                    if (returnStatus || exitCode == 0) {
+                        getContext().onSuccess(returnStatus ? exitCode : returnStdout ? new String(controller.getOutput(workspace, launcher), encoding) : null);
                     } else {
                         getContext().onFailure(new AbortException("script returned exit code " + exitCode));
```

Comment:
I guess someone configuring `returnStdout` expects to get the stdout even when the step fails, but in the exception message maybe...

Belated :bug: I guess